crowbar_framework/app/models/backup.rb

```diff
@@ -178,16 +178,28 @@ def save_archive
       return false
     end
 
-    path.open("wb") do |f|
-      f.write(file.read)
+    meta_file = data.join("meta.yml")
+    unless meta_file.exist?
+      errors.add(:file_content, I18n.t("backups.index.meta_missing"))
+      return false
+    end
+
+    begin
+      meta = YAML.load_file(meta_file)
+    rescue Psych::SyntaxError
+      errors.add(:file_content, I18n.t("backups.index.invalid_file_content"))
+      return false
     end
 
-    meta = YAML.load_file(data.join("meta.yml"))
     self.version = meta["version"].to_s
     self.size = path.size
     self.created_at = Time.zone.parse(meta["created_at"])
     # 20151222144602_create_backups.rb
     self.migration_level = meta["migration_level"] || 20151222144602
+
+    path.open("wb") do |f|
+      f.write(file.read)
+    end
```

Comment:
Argh, I think I have just discovered a flaw here, we have to write the file before the `meta_file.exist?` check...

as we call `data` which calls `extract` which uses `path` which is empty at that time because we haven't written it yet... so after all we need to write the file **before** performing the validation... otherwise we will always run into `backups.index.meta_missing`

we could write it before the validation, and in case of an error delete it again